test/system/test_server_tls.hpp

```diff
@@ -7,29 +7,30 @@
 #if !defined(MQTT_TEST_SERVER_TLS_HPP)
 #define MQTT_TEST_SERVER_TLS_HPP
 
+#if defined(MQTT_USE_TLS)
 #include <set>
 
 #include <boost/lexical_cast.hpp>
 #include <boost/multi_index_container.hpp>
 #include <boost/multi_index/ordered_index.hpp>
 #include <boost/multi_index/member.hpp>
 #include <boost/multi_index/identity.hpp>
+#include <boost/asio.hpp>
+#include <boost/asio/ssl.hpp>
 
 #include <mqtt_server_cpp.hpp>
-#include "test_settings.hpp"
-#include "test_ctx_init.hpp"
 
 namespace mi = boost::multi_index;
 namespace as = boost::asio;
 
-class test_server_tls : ctx_init {
+class test_server_tls {
 public:
-    test_server_tls(as::io_context& ioc, MQTT_NS::broker::broker_t& b)
+    test_server_tls(as::io_context& ioc, boost::asio::ssl::context&& ctx, MQTT_NS::broker::broker_t& b, uint16_t port = broker_tls_port)
         : server_(
             as::ip::tcp::endpoint(
-                as::ip::tcp::v4(), broker_tls_port
+                as::ip::tcp::v4(), port
             ),
-            std::move(ctx),
+            std::forward<boost::asio::ssl::context>(ctx),
```

Comment:
```suggestion
            MQTT_NS::force_move(ctx),
```

ctx is rvalue-reference not forwarding reference in your code.